provider/common/cluster_waiter.go

```diff
@@ -47,15 +47,23 @@ func (dw *DefaultClusterWait) WaitForClusterToBeReady(ctx context.Context, clust
 	resource := dw.collection.Cluster(clusterId)
 	// We expect the cluster to already exist
 	// Try to get it and if result with NotFound error, return error to user
-	if resp, err := resource.Get().SendContext(ctx); err != nil && resp.Status() == http.StatusNotFound {
-		message := fmt.Sprintf("Cluster %s not found, error: %v", clusterId, err)
+	resp, err := resource.Get().SendContext(ctx)
```

Comment:
since this function is a duplication (or a specific case ) of RetryClusterReadiness, can you refactor the code to NOT use `WaitForClusterToBeReady` at all and use `RetryClusterReadiness`

Reply:
@robpblake  it's fine for me to approve it and fix it in a new PR as a refactoring for several resources